pkg/remotewrite/validate_test.go

```diff
@@ -117,3 +118,114 @@ func TestFindClusterIDs(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateRequest(t *testing.T) {
+	type args struct {
+		remoteWriteRequest *prometheus.WriteRequest
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "should return an error if there are multiple cluster IDs in the request",
+			args: args{
+				remoteWriteRequest: reqWithMultipleClusterID,
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error if there are no cluster IDs in the request",
+			args: args{
+				remoteWriteRequest: req,
+			},
+			wantErr: true,
+		},
+		{
+			name: "should succeed if there is one cluster ID in the request",
+			args: args{
+				remoteWriteRequest: reqWithOneClusterID,
+			},
+			want:    "cluster-1",
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := ValidateRequest(tt.args.remoteWriteRequest)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ValidateRequest() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("ValidateRequest() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestIsMetadataRequest(t *testing.T) {
+	metadataReq := &prometheus.WriteRequest{
```

Comment:
Is there some reason the variables have been created first? Can we do the same but in each `args` value?